src/Shared/PackageManagers/BaseProjectManager.cs

```diff
@@ -18,19 +18,25 @@ namespace Microsoft.CST.OpenSource.PackageManagers
     using Version = SemanticVersioning.Version;
     using PackageUrl;
 
-    public class BaseProjectManager
+    public abstract class BaseProjectManager
     {
+        /// <summary>
+        /// The type of the project manager from the package-url type specifications.
+        /// </summary>
+        /// <seealso href="https://www.github.com/package-url/purl-spec/blob/master/PURL-TYPES.rst"/>
+        public static readonly string Type = null!;
```

Comment:
```suggestion
        public static readonly string Type = "Undefined";
```

Recommend not forcing this to be null.  Its still possible to access static properties of abstract classes and returning null here breaks the contract that Type is a non-null string.  No strong preference on what value to use here, `string.Empty` would be fine, `Undefined`, `Invalid` are all options that seems fine. Unfortunately I didn't see any already defined value for this type of invalid scenario in the types you posted from the purl repo.

Reply:
Changed to `undefined`